src/Chapter2.hs

```diff
@@ -620,7 +650,22 @@ Write a function that takes elements of a list only on even positions.
 >>> takeEven [2, 1, 3, 5, 4]
 [2,3,4]
 -}
-takeEven = error "takeEven: Not implemented!"
+
+-- takeEven :: [a] -> [a]
+-- takeEven [] = [] 
+-- takeEven [x] = [x]
+-- takeEven (x:_:xs) = x : takeEven xs
+
+-- takeEven :: [a] -> [a]
+-- takeEven [] = [] 
+-- takeEven (x:xs) = x : takeEven (drop 1 xs)
+
+takeEven :: [a] -> [a]
+takeEven list = go [] list
+    where 
+        go :: [a] -> [a] -> [a]
+        go acc [] = acc
+        go acc (x:xs) = go (acc ++ [x]) (drop 1 xs)
```

Comment:
Usually, adding elements to the end of the list is costly. Your implementation is correct :+1: But you can try to implement this function more efficiently by using `:` instead `++`. If you can think, how to avoid adding elements to the end of the list, and instead add to the beginning, but get the same result, that would be amazing!

In general, it's always a good idea to try to replace `l ++ [x]` with `x : l` whenever possible, though, in some cases it might not be that trivial to achieve the same result. Remember the analogy with trains :bullettrain_side: :train: :train: :train: